kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -140,6 +141,10 @@ public void createTopic(
       @Suspended AsyncResponse asyncResponse,
       @PathParam("clusterId") String clusterId,
       @Valid CreateTopicRequest request) {
+
+    if (request == null) {
+      throw Errors.invalidPayloadException("Request body is empty. Data is required.");
```

Comment:
We should make the error message match what is in ProduceAction.  There it says:

```
    if (requests == null) {
      throw Errors.invalidPayloadException("Null input provided. Data is required.");
    }
```

And now I'm debating whether these two errors should return a 404 rather than a 422 (The 422 was chosen for ProduceAction as that matches the other cases of malformed data in the class).  

But as long as the two methods return the same error code I don't have too strong feelings.

Have you taken a look at any of our other POST endpoints (eg createACL) to see what they do with no data provided?

Reply:
Agreed, 422 seems like a slightly worse fit, and also is less well understood 4xx response status code compared to the ubiquitous 404.